### DISCUSSION TITLE
Is it possible to parse wikilink-style citekeys when exporting from markdown using the zotero lua filter?

### DISCUSSION THREAD

**Discussion by simonvanlierde** [retorquere/zotero-better-bibtex#2844] [Q&A]
Hello everyone!

I'm utilizing the [Better Bibtex Zotero Lua Filter](https://retorque.re/zotero-better-bibtex/exporting/zotero.lua) to convert citations from the `[@citekey]` format into live Zotero citations while exporting markdown files from Obsidian to Word via Pandoc. This setup works well for standard citations, but I'm struggling to adapt it for wikilink-style citations, specifically:

`[[@citekey]]` (double brackets)
`[[@citekey | link-name]]`

To address this, I attempted to preprocess these wikilinks with another Lua filter that converts them to a single-bracket format suitable for the Zotero filter:
```lua
function Inlines(inlines)
    local newInlines = {}
    for _, inline in ipairs(inlines) do
        if inline.t == 'Link' then
            local citekey = inline.target:match("^(@.+)$")
            table.insert(newInlines, "[" .. citekey .. "]")
        else
            table.insert(newInlines, inline)
        end
    end
    return newInlines
end
```
However, this approach does not result in them being recognized as live Zotero citations in the exported DOCX file—they simply appear as plain text within single brackets (`[@citekey]`)

Does anyone have experience or suggestions on how to modify the Zotero Lua filter to recognize and convert these wikilink-style citations into live Zotero citations in Word documents? Any insights or guidance would be greatly appreciated!

**Comment by retorquere** [comment]
If you call `pandoc --from=markdown --to=native source-doc.md`, you get a dump of how pandoc parses your document. You will have to insert or modify `Cite` records, as that is what the BBT lua filter looks for.

**Reply by retorquere** [reply]
Something like this should get you started:

```
function Link(el)
  -- Check if the link is a wikilink-style citation
  if string.match(el.target, "^%[%[.*%]%]$") then
    -- Remove the brackets to get the citation key
    local citationKey = string.gsub(el.target, "%[%[(.*)%]%]", "%1")
    -- Create a new citation
    local citation = pandoc.Cite(
      {}, -- Mode (NormalCitation, AuthorInText, SuppressAuthor)
      {pandoc.Citation(citationKey, "NormalCitation", 0, 0, "")} -- Citations
    )
    -- Replace the link with the citation
    return citation
  end
  -- If it's not a wikilink-style citation, leave it unchanged
  return el
end
```

**Reply by simonvanlierde** [reply]
This was helpful, thank you!
